core/primitives/src/subtrie.rs

```diff
@@ -0,0 +1,293 @@
+// Copyright 2017-2019 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Child trie related struct
+
+use parity_codec::{Encode, Decode};
+use rstd::prelude::*;
+use crate::storage::well_known_keys::CHILD_STORAGE_KEY_PREFIX;
+#[cfg(feature = "std")]
+pub use impl_serde::serialize as bytes;
+
+/// keyspace type.
+pub type KeySpace = Vec<u8>;
+
+
+/// info related to parent trie.
+/// Full key of child trie storage location
+/// and size of the prefix of this location.
+pub type ParentTrie = Vec<u8>;
```

Comment:
not at all clear how a `Vec<u8>` could be used to express a "Full key" and a "size".